packages/contracts-bedrock/src/dispute/AnchorStateRegistry.sol

```diff
@@ -23,105 +22,222 @@ import { ISuperchainConfig } from "interfaces/L1/ISuperchainConfig.sol";
-    mapping(GameType => OutputRoot) public anchors;
+    // TODO: storage slot issue
+    // / @notice Returns the anchor state for the given game type.
+    // mapping(GameType => OutputRoot) public anchors;
+
+    // TODO: determine if this belongs
+    // uint256 __gap0;
 
     /// @notice Address of the SuperchainConfig contract.
     ISuperchainConfig public superchainConfig;
 
-    /// @param _disputeGameFactory DisputeGameFactory address.
-    constructor(IDisputeGameFactory _disputeGameFactory) {
-        DISPUTE_GAME_FACTORY = _disputeGameFactory;
+    /// @notice Returns whether a game is blacklisted.
+    mapping(IDisputeGame => bool) public isGameBlacklisted;
+
+    uint64 public gameRetirementTimestamp;
+
+    GameType public respectedGameType;
+
+    IFaultDisputeGame[] public anchorGameCandidates;
+
+    uint256 internal _anchorGameCandidateIndex;
+
+    IFaultDisputeGame internal _anchorGame;
+
+    uint256 public tryUpdateAnchorStateGas;
+
+    constructor() {
         _disableInitializers();
     }
 
     /// @notice Initializes the contract.
-    /// @param _startingAnchorRoots An array of starting anchor roots.
+    /// @param _disputeGameFactory DisputeGameFactory address.
+    /// @param _disputeGameFinalityDelaySeconds The delay between when a dispute game is resolved and when it can be
+    /// considered finalized.
+    /// @param _authorizedGame An authorized dispute game.
     /// @param _superchainConfig The address of the SuperchainConfig contract.
+    /// @param _tryUpdateAnchorStateGas The approximate gas limit for the tryUpdateAnchorState loop.
     function initialize(
-        StartingAnchorRoot[] memory _startingAnchorRoots,
-        ISuperchainConfig _superchainConfig
+        IDisputeGameFactory _disputeGameFactory,
+        uint256 _disputeGameFinalityDelaySeconds,
+        IFaultDisputeGame _authorizedGame,
+        ISuperchainConfig _superchainConfig,
+        uint256 _tryUpdateAnchorStateGas
     )
         external
         initializer
     {
-        for (uint256 i = 0; i < _startingAnchorRoots.length; i++) {
-            StartingAnchorRoot memory startingAnchorRoot = _startingAnchorRoots[i];
-            anchors[startingAnchorRoot.gameType] = startingAnchorRoot.outputRoot;
-        }
+        disputeGameFactory = _disputeGameFactory;
+        disputeGameFinalityDelaySeconds = _disputeGameFinalityDelaySeconds;
+        _anchorGame = _authorizedGame;
         superchainConfig = _superchainConfig;
+        tryUpdateAnchorStateGas = _tryUpdateAnchorStateGas;
     }
 
-    /// @notice Returns the DisputeGameFactory address.
-    /// @return DisputeGameFactory address.
-    function disputeGameFactory() external view returns (IDisputeGameFactory) {
-        return DISPUTE_GAME_FACTORY;
+    /// @notice Returns the output root of the anchor game, or an authorized anchor state if no such game exists.
+    function anchors(GameType /* unused */ ) public view returns (OutputRoot memory) {
```

Comment:
trying to be mindful of ABI changes. in the spec we call this `getAnchorState`. also, we don't need GameType anymore. Is it worth examining the dependency graph here for breaking ABI change? assumed not.